### ISSUE TITLE
Updating patient information via API

### ISSUE THREAD

**Issue by voidspacexyz** [openhealthcare/opal#1274]
I am trying to update the demographic of a Patient using POSTMAN / Curl etc.

Here is what I am doing in POSTMAN: 
http://myip:port/api/v0.1/demographics/5/

Headers : 
Authorization : Token my-token-goes-here
Content-Type: application/json

Request Body

first_name : Some Name

Now the first time I run it on any of the patient, it return me this output.

{
  "consistency_token": "8bd857cd",
  "surname": "Wilson",
  "sex": "Not Specified",
  "birth_place": "",
  "id": 5,
  "ethnicity": "",
  "nhs_number": "1667729",
  "first_name": "Olivia",
  "middle_name": "Changed123",
  "title": "",
  "religion": null,
  "death_indicator": false,
  "updated": "26/10/2017 12:35:22",
  "post_code": null,
  "date_of_birth": "05/08/1997",
  "gp_practice_code": null,
  "date_of_death": null,
  "hospital_number": "1667729",
  "created_by_id": null,
  "created": null,
  "marital_status": "",
  "patient_id": 5,
  "updated_by_id": 1
}

But now if I try to update something else, lets say middle_name on the same patient, it gives me this error.

400 OK
{
  "error": "Unexpected field name"
}

I initially thought I was doing something wrong, but then how can it work the first time alone and not after that ? I am a little puzzled here.

PS : Same updating via the GUI works just fine. I only have a problem with API calls directly.

**Comment by fredkingham** [comment]
sounds annoying, I ran

curl http://127.0.0.1:8000/api/v0.1/past_medical_history/  -d '{"updated": null, "consistency_token": "a3dfe1d9", "created": "27/10/2017 12:40:00", "episode_id": 28, "updated_by_id": null, "details": "asdd", "year": "as", "created_by_id": 1, "condition": "AKA"}' -H 'Content-Type: application/json' -H 'Authorization: Token 4fc401fc2e7508b455a6972917e2da6b1e4fbbc0'

and that worked fine
(obviously replace your own token)

The I ran a put with the below (updating the year)

curl -X PUT http://127.0.0.1:8000/api/v0.1/past_medical_history/2/  -d '{"updated": null, "consistency_token": "7b2ea135", "created": "27/10/2017 12:40:00", "episode_id": 28, "updated_by_id": null, "details": "asdd", "year": "2017", "created_by_id": 1, "condition": "AKA"}' -H 'Content-Type: application/json' -H 'Authorization: Token 4fc401fc2e7508b455a6972917e2da6b1e4fbbc0'

and that worked fine.

I think its most likely occurred because, maybe you didn't update the consistency token.

We expect a consistency token with every 'PUT' request, this is so multiple clinicians don't accidentally try and update the same subrecord at the same time.  

I just tried posting without the consistency token, and you get the error you've mentioned. I think we could make this clearer. 

Let me know if that's it

I've raised the vague error message as an issue here https://github.com/openhealthcare/opal/issues/1279
